### PR TITLE
Fixes forecast issues

### LINKED ISSUE #366: flood_forecast_risk returns netcdf with non-descriptive variable name
The function returns a netcdf with a variable named "q_sim" whereas it should be "exceedance_probability" or something similar.

### PR DESCRIPTION
**Pull request by richardarsenault**
## Overview

This PR fixes the problems with hindcasting notebooks, by creating one that uses Ravenpy rather than the WPS-based one we currently use. Fixes #366 and #359.

Changes:

- Added Notebook "12_Performing_hindcasting_experiments" (note that it requires the "input.geojson" shape file that is generated in a previous tutorial notebook, so running it standalone will fail)
- Fixed the GEPS forecast precipitation scaling so hindcasts are now realistic
- Fixed the naming scheme of flood forecast risk output to make it more descriptive.

### CODE REVIEW

File: raven/processes/wps_forecast_flood_risk.py [hunk 1]
```diff
@@ -99,7 +99,9 @@ def _handler(self, request, response):
             )  # This is needed to return values instead of floats
 
         # Put in file
+
         filename = tempfile.mkdtemp() + "/fcst_flood_risk.nc"
+        pct.name = "exceedance_probability"
```
Comment (huard) [inline raven/processes/wps_forecast_flood_risk.py:104]: Here you're converting a DataArray to a Dataset, but the Dataset will have no global attributes. 
I suggest doing something like
```
out = pct.to_dataset(name="exceedance_probability")
out.attrs = ...
out.to_netcdf(filename)
```
where you copy attributes from the original file, or create new attributes describing the file content. You probably minimally want attributes like title, source and history.

**Review by huard — APPROVED** [review]

File: docs/source/notebooks/Assess_probabilistic_flood_risk.ipynb [hunk 2]
```diff
@@ -12,15 +12,10 @@
   },
```
Comment (huard) [inline docs/source/notebooks/Assess_probabilistic_flood_risk.ipynb:12]: <pre class="ql-syntax" spellcheck="false">Line #6.    resp_fa = finch.(
</pre><p>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/raven/pull/413/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Comment by huard** [comment]
J'ai laissé d'autres commentaires dans ReviewNB qui n'apparaissent pas ici.

**Comment by richardarsenault** [comment]
@Zeitsperre It seems linting is failing, and I think you have applied the "black" formatting to this to make it pass? Can you work your magic here too, or show me the ropes so I can do it myself? Thanks!

**Comment by Zeitsperre** [comment]
@richardarsenault 
```
$ conda install -c conda-forge pre_commit
$ pre-commit install
$ pre-commit run --all-files
```
The process is fully automated after that. All checks will run every time you commit on the files that you change. It's magic!

**Comment by richardarsenault** [comment]
Well, that was easy! Thanks!

**Pull request merged**